cmd/cluster/core/create.go

```diff
@@ -587,14 +587,22 @@ type Platform interface {
 }
 
 func CreateCluster(ctx context.Context, opts *CreateOptions, platform Platform) error {
-	if opts.Wait && opts.NodePoolReplicas < 1 {
-		return errors.New("--wait requires --node-pool-replicas > 0")
+	if len(opts.InfraID) == 0 {
+		opts.InfraID = infraid.New(opts.Name)
+	}
+
+	if err := Validate(ctx, opts); err != nil {
+		return err
 	}
 
 	if err := platform.Complete(ctx, opts); err != nil {
 		return fmt.Errorf("could not complete platform specific options: %w", err)
 	}
 
+	if err := platform.Validate(ctx, opts); err != nil {
```

Comment:
Sorry for missing this - generally the validation expects to run before completion, could we hoist this please? I'll follow up with a Go type trick to make it impossible to make this mistake after your PR goes in.